qiskit_optimization/applications/ising/max_cut.py

```diff
@@ -1,83 +1,38 @@
-# that they have been altered from the originals.
-
-"""
-Convert max-cut instances into Pauli list
-Deal with Gset format. See https://web.stanford.edu/~yyye/yyye/Gset/
-Design the max-cut object `w` as a two-dimensional np.array
-e.g., w[i, j] = x means that the weight of a edge between i and j is x
-Note that the weights are symmetric, i.e., w[j, i] = x always holds.
-"""
-
-from typing import Tuple
-import logging
-
-import numpy as np
-
-from qiskit.quantum_info import Pauli
-from qiskit.opflow import PauliSumOp
-
-logger = logging.getLogger(__name__)
-
-
-def get_operator(weight_matrix: np.ndarray) -> Tuple[PauliSumOp, float]:
-    """Generate Hamiltonian for the max-cut problem of a graph.
-
-    Args:
-        weight_matrix: adjacency matrix.
-
-    Returns:
-        operator for the Hamiltonian
-        a constant shift for the obj function.
-
-    """
-    num_nodes = weight_matrix.shape[0]
-    pauli_list = []
-    shift = 0
-    for i in range(num_nodes):
-        for j in range(i):
-            if weight_matrix[i, j] != 0:
-                x_p = np.zeros(num_nodes, dtype=bool)
-                z_p = np.zeros(num_nodes, dtype=bool)
-                z_p[i] = True
-                z_p[j] = True
-                pauli_list.append([0.5 * weight_matrix[i, j], Pauli((z_p, x_p))])
-                shift -= 0.5 * weight_matrix[i, j]
-    opflow_list = [(pauli[1].to_label(), pauli[0]) for pauli in pauli_list]
-    return PauliSumOp.from_list(opflow_list), shift
-
-
-def max_cut_value(x, w):
-    """Compute the value of a cut.
-
-    Args:
-        x (numpy.ndarray): binary string as numpy array.
-        w (numpy.ndarray): adjacency matrix.
-
-    Returns:
-        float: value of the cut.
-    """
-    # pylint: disable=invalid-name
-    X = np.outer(x, (1 - x))
-    return np.sum(w * X)
-
-
-def get_graph_solution(x):
-    """Get graph solution from binary string.
-
-    Args:
-        x (numpy.ndarray) : binary string as numpy array.
-
-    Returns:
-        numpy.ndarray: graph solution as binary numpy array.
-    """
-    return 1 - x
+from functools import lru_cache
+
+import networkx as nx
+from docplex.mp.model import Model
+from qiskit_optimization.problems.quadratic_program import QuadraticProgram
+
+from .graph_application import GraphApplication
+
+
+class Maxcut(GraphApplication):
+
+    def __init__(self, g):
+        super().__init__(g)
+
+    def _build_quadratic_program(self):
+        mdl = Model(name='maxcut')
+        x = {i: mdl.binary_var(name='x_{0}'.format(i)) for i in range(self._g.number_of_nodes())}
+        for u, v in self._g.edges:
+            self._g.edges[u, v].setdefault('weight', 1)
+        objective = mdl.sum(self._g.edges[i, j]['weight'] * x[i]
+                            * (1 - x[j]) for i, j in self._g.edges)
+        mdl.maximize(objective)
+        qp = QuadraticProgram()
+        qp.from_docplex(mdl)
+        return qp
+
+    def plot_graph(self, x, pos=None):
```

Comment:
`x` seems such a generic name when looking at this method. From the notebook sample this seems to be `x` from the result (or some other equivalent value of your choosing that may not be the result). Would it make sense to be able to pass the Result object here or `x` such that its easy to pass the result, which seems to be more the usual use case to me such that x is taken from it, as well as passing one more directly to try other values.